### PR TITLE
Improving data onboarding flow across Action Network, Google Sheets and Airtable [MAP-670] [MAP-671]

### PR DESCRIPTION
**Pull request by Moggach**
## Description
This PR adds the following:
1. Airtable Base and Table IDs are extracted from the url for the Airtable base
2. Action Network group slug is extracted from the url for the Action Network group
3. Google Sheets id is extracted from  the Google Sheets URL
4. Google Sheets connected fetches the sheet names and puts them in a dropdown

## Motivation and Context
Addresses issues [MAP-671](https://linear.app/commonknowledge/issue/MAP-671/auto-extract-sheet-id-and-sheet-name-for-google-sheet) and [MAP-670](https://linear.app/commonknowledge/issue/MAP-670/auto-extract-group-slug-from-action-network-url)

## How Can It Be Tested?
Run the branch locally
Go through the add data source flows for Airtable, Action Network and Google Sheets
To authenticate a Google Sheet you'll need to run the frontend server with `npm run https`

### CODE REVIEW

File: nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx [hunk 1]
```diff
@@ -406,23 +468,19 @@ export default function Page({
...
 
   useEffect(() => {
     if (airtableUrl) {
       try {
-        const url = new URL(airtableUrl)
-        const [_, base, table, ...pathSegments] = url.pathname.split('/')
-        form.setValue('airtable.baseId', base)
-        form.setValue('airtable.tableId', table)
+        const { baseId, tableId } = extractBaseAndTableId(airtableUrl)
+        if (baseId) form.setValue('airtable.baseId', baseId)
```
Comment (joaquimds) [inline nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx:476]: if statements without curly bracers `{ around_the_body; }` give me _the fear_

**Review by joaquimds — APPROVED** [review]

File: nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx [hunk 2]
```diff
@@ -1210,22 +1231,51 @@ export default function Page({
...
+
+                            setLoadingSheets(false)
+                          }
+                        } catch (err) {
+                          setFetchError(
+                            'Failed to fetch sheet names. Please check the URL or credentials.'
+                          )
+                          setLoadingSheets(false)
+                        }
+                      }}
```
Comment (janbaykara) [inline nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx:1273]: Looks good but can we define this function elsewhere so the UI bit of the file is kept tidy JSX? Inspired by @ev-sc's recent changes to the map code.

File: nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx [hunk 3]
```diff
@@ -205,6 +206,67 @@ export default function Page({
...
+        return { baseId, tableId }
+      }
+      return {}
+    } catch (error) {
+      console.error('Error extracting Base ID and Table ID:', error)
+      return {}
+    }
+  }
+
+  function extractGroupSlug(url: string): string | null {
```
Comment (janbaykara) [inline nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx:260]: `extractActionNetworkGroupSlug` perhaps?

**Comment by janbaykara** [comment]
LGTM bar comments

File: nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx (hunk 2 above)
Comment (Moggach) [reply nextjs/src/app/(logged-in)/data-sources/create/connect/[externalDataSourceType]/page.tsx]: good shout, done

**Pull request merged**